### DISCUSSION TITLE
In a flow, set a condition to detect is a user exists from it's email

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#23765] [Feature Requests]
### Summary

The Options is

```json
{
  "message": [
    {
      "id": "0abb92df-d1dc-4ed0-abdd-ab82a27e01ce",
      "email": "myaddress@gmail.com"
    }
  ]
}
```

nothing worked to detect if $last is empty or not. For example:

```json
{
    "$last[0]": {
        "id": {
            "_nnull": true
        }
    }
}

```

says 

```json
[
  {
    "name": "ValidationError",
    "message": "\"$last[0]\" is required",
    "stack": "ValidationError: \"$last[0]\" is required"
  }
]
```

### Motivation

-

### Detailed Design

-

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

- Time!

### Alternatives

Not using flow

### Adoption Strategy

-

**Comment by domdus** [comment]
There is no way to check array data (`$last.message`) with a `condition operation`.

**1. Option: Use Run Script**
```
module.exports = async function(data) {
    if (data.$last.message.length > 0) {
        // We have a user, return true -> return {}
	return data.$last.message[0]
    } else {
        // Throw error to proceed with false case
        throw new Error("No User")
    }
}
```

**2. Option: Pass to another flow**
`"message": []` is an array.
In directus flows, working with array items means you need to pass it to another flow -> `trigger another flow operation` and pass `{{$last.message}}` to loop through each item. (also if just one inside)
- If the array is empty (no user?), this 2nd flow will never execute
- If array contains a user, 2nd flow will execute and you can work with the data there
